docs/basics/networkvariable.md

```diff
@@ -504,6 +504,51 @@ In order to create your own `NetworkVariableBase` derived container, you should:
     - `void ReadDelta(FastBufferReader reader, bool keepDirtyDelta)`
 - Depdending upon your custom `NetworkVariableBase` container, you might look at `NetworkVariable<T>` or `NetworkList` to see how those two examples were implemented.
 
+#### NetworkVariableSerialization&lt;T&gt;
+
+When working with known, non-generic types, it is easy and efficient to read and write using `FastBufferReader.ReadValue` and `FastBufferWriter.WriteValue`, respectively, or for integer types, using `BytePacker` and `ByteUnpacker` to add compression. When working with generic types, however, we have to rely on serializers that are generated based on types discovered during a compile-time code generation process. As such, when working with generic types, you have to a little bit of extra work to tell our code generation algorithm which types to generate serializers for. You can do this in two ways:
+
+The first way is to use `GenerateSerializationForTypeAttribute` to generate serialization for a specific hard-coded type:
+```csharp
+[GenerateSerializationForType(typeof(Foo))]
+public class MyNetworkVariableTypeUsingFoo : NetworkVariableBase {}
+```
+
+However, this is rarely useful for generic types, since if you know the type, you can generally use the more efficient route of calling `FastBufferReader`/`FastBufferWriter` methods directly.
```

Comment:
That statement is poorly worded, sorry... This will answer your other question below, too.

In a generic type, you have `GenericType<T>` where T is a generic type parameter (an unknown type).

When you do `[GenerateSerializationForType(typeof(Foo))]` what that does is it allows you to call `NetworkVariableSerialization<Foo>.Write()` and `NetworkVariableSerialization<Foo>.Read()`.

But if you know the type you are using is `Foo`, as opposed to the generic type parameter `T` which is an unknown type, you can usually just call `FastBufferWriter.WriteValue()` and `FastBufferReader.ReadValue()`, which do not need that kind of codegen-based initialization and are generally more efficient.

But if you want to serialize that value of type `T`, you can't call methods on `FastBufferWriter` and `FastBufferReader` because `T` is an unknown type and those types can't handle that. `NetworkVariableSerialization<T>` is a higher-level concept that uses codegen to determine how to serialize unknown types like that, but the codegen has to know what types to generate for.

with `[GenerateSerializationForGenericParameter()]` the codegen does some magic to figure out what `T` is and generate serialization for it, which is useful.

`[GenerateSerializationForType()]` is for if, for some reason, you know the type, but you want to call `NetworkVariableSerialization<T>` methods on it instead of `FastBufferReader`/`FastBufferWriter` methods. In 1.8.0 there isn't a whole lot of value in that - I can't think of an actual use case, but that doesn't mean there isn't one, so I left this there just in case.

In 1.9.0 there will be methods added for doing delta serialization of the differences between a current value and a previous one, so this will be more useful then.

So with that in mind... it may be best to just remove the comments about the usefulness of these methods entirely.

Reply:
This explination helped _a lot_ thank you!
I've made a small change to the succestion that hopefully makes this clearer, but if you'd still prefer to remove it that's okay.